libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewOfflineRouter.java

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import com.mapbox.services.android.navigation.v5.navigation.MapboxOfflineRouter;
+import com.mapbox.services.android.navigation.v5.navigation.NavigationRoute;
+import com.mapbox.services.android.navigation.v5.navigation.OfflineRoute;
+
+class NavigationViewOfflineRouter {
+
+  private final MapboxOfflineRouter offlineRouter;
+  private final NavigationViewRouter router;
+  private boolean isConfigured;
+  private String tileVersion;
+
+  NavigationViewOfflineRouter(MapboxOfflineRouter offlineRouter, NavigationViewRouter router) {
+    this.offlineRouter = offlineRouter;
+    this.router = router;
+  }
+
+  void configure(String tileVersion) {
+    if (!isConfigured || isNew(tileVersion)) {
+      offlineRouter.configure(tileVersion, new OfflineRouterConfiguredCallback(this));
+    }
+    this.tileVersion = tileVersion;
+  }
+
+  void setIsConfigured(boolean isConfigured) {
+    this.isConfigured = isConfigured;
+  }
+
+  boolean isConfigured() {
+    return isConfigured;
+  }
+
+  void findRouteWith(NavigationRoute.Builder builder) {
+    if (!isConfigured) {
+      return;
```

Comment:
Should we give an error if not configured?

Reply:
Yeah, good catch

Reply:
Should we throw an exception here? As it is, this would be a somewhat silent failure.

Reply:
We actually won't call this method if this class isn't configured https://github.com/mapbox/mapbox-navigation-android/pull/1829/files#diff-29009b5750e50557cd5e8a7fd79a46f7R80.  So we may not need this check at all?

Reply:
Yeah, let's remove it altogether then.